### ISSUE TITLE
training issue

### ISSUE THREAD

**Issue by popov1212** [YvanYin/GoalFlow#32]
Hi, Thanks for your excellent work.
I tried to run the train script as follows with updated code:

export OPENSCENE_DATA_ROOT=/data/navsim_workspace/dataset  # 根据您的实际数据路径调整
export NAVSIM_EXP_ROOT=/data/navsim_workspace/exp         # 根据您的实际实验路径调整
export NUPLAN_MAPS_ROOT=/data/navsim_workspace/dataset/maps
export PYTHONPATH=/data/workspace/GoalFlow:/data/navsim_workspace/navsim:$PYTHONPATH

FEATURE_CACHE='/data/navsim_workspace/exp/goalflow_trainvalcache' # set your feature_cache path
V99_PRETRAINED_PATH='/data/workspace/GoalFlow/data/depth_pretrained_v99-3jlw0p36-20210423_010520-model_final-remapped.pth'
CHECKPOINT_PATH=/data/workspace/GoalFlow/data/goalflow_traj_epoch_54-step_18260.ckpt
VOC_PATH='/data/workspace/GoalFlow/data/cluster_points_8192_.npy'
ONLY_PERCEPTION=False
FREEZE_PERCEPTION=True # you can choose False and increase batch_size if the GPU are sufficient

python /data/workspace/GoalFlow/navsim/planning/script/run_training.py \
agent=goalflow_agent_traj \
experiment_name=a_train_traj \
scene_filter=navtrain \
split=trainval \
cache_path=$FEATURE_CACHE \
trainer.params.max_epochs=100 \
agent.config.training=True \
agent.config.has_navi=True \
agent.config.start=True \
agent.config.freeze_perception=$FREEZE_PERCEPTION \
agent.config.only_perception=$ONLY_PERCEPTION \
agent.config.train_scale=0.1 \
agent.config.tf_d_model=1024 \
agent.config.trajectory_weight=50.0 \
agent.config.agent_class_weight=0.2 \
agent.config.agent_box_weight=0.05 \
agent.config.bev_semantic_weight=0.2 \
agent.config.agent_loss=True \
dataloader.params.batch_size=2 \
use_cache_without_dataset=True \
agent.config.v99_pretrained_path=$V99_PRETRAINED_PATH \
agent.config.voc_path=$VOC_PATH

but the loss seems weird as follows:

<img width="2054" height="1027" alt="Image" src="https://github.com/user-attachments/assets/0097b648-353d-4297-a60c-4a891af27331" />
the picture shows above is in the No.0 epoch, but the trend of loss curve seems same as above in No.20 epoch. I tried to modified the batch_size, seems not work. Also, I tried to remove/add "agent.checkpoint_path=$CHECKPOINT_PATH", but seems not work as well.

by the way, I use 2*5090 for training. And the files are download for your relased google
drive, including "depth_pretrained_v99-3jlw0p36-20210423_010520-model_final-remapped.pth", "goalflow_traj_epoch_54-step_18260.ckpt", "cluster_points_8192_.npy".

could you help me to fingure out what is the cause of this issue? Thanks in ahead for your kindly reply!

**Comment by fengjiang5** [comment]
> Hi, Thanks for your excellent work. I tried to run the train script as follows with updated code:
> 
> export OPENSCENE_DATA_ROOT=/data/navsim_workspace/dataset # 根据您的实际数据路径调整 export NAVSIM_EXP_ROOT=/data/navsim_workspace/exp # 根据您的实际实验路径调整 export NUPLAN_MAPS_ROOT=/data/navsim_workspace/dataset/maps export PYTHONPATH=/data/workspace/GoalFlow:/data/navsim_workspace/navsim:$PYTHONPATH
> 
> FEATURE_CACHE='/data/navsim_workspace/exp/goalflow_trainvalcache' # set your feature_cache path V99_PRETRAINED_PATH='/data/workspace/GoalFlow/data/depth_pretrained_v99-3jlw0p36-20210423_010520-model_final-remapped.pth' CHECKPOINT_PATH=/data/workspace/GoalFlow/data/goalflow_traj_epoch_54-step_18260.ckpt VOC_PATH='/data/workspace/GoalFlow/data/cluster_points_8192_.npy' ONLY_PERCEPTION=False FREEZE_PERCEPTION=True # you can choose False and increase batch_size if the GPU are sufficient
> 
> python /data/workspace/GoalFlow/navsim/planning/script/run_training.py agent=goalflow_agent_traj experiment_name=a_train_traj scene_filter=navtrain split=trainval cache_path=$FEATURE_CACHE trainer.params.max_epochs=100 agent.config.training=True agent.config.has_navi=True agent.config.start=True agent.config.freeze_perception=$FREEZE_PERCEPTION agent.config.only_perception=$ONLY_PERCEPTION agent.config.train_scale=0.1 agent.config.tf_d_model=1024 agent.config.trajectory_weight=50.0 agent.config.agent_class_weight=0.2 agent.config.agent_box_weight=0.05 agent.config.bev_semantic_weight=0.2 agent.config.agent_loss=True dataloader.params.batch_size=2 use_cache_without_dataset=True agent.config.v99_pretrained_path=$V99_PRETRAINED_PATH agent.config.voc_path=$VOC_PATH
> 
> but the loss seems weird as follows:
> 
> <img alt="Image" width="2000" height="1027" src="https://private-user-images.githubusercontent.com/34418403/487083934-0097b648-353d-4297-a60c-4a891af27331.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgwMTEyMzYsIm5iZiI6MTc1ODAxMDkzNiwicGF0aCI6Ii8zNDQxODQwMy80ODcwODM5MzQtMDA5N2I2NDgtMzUzZC00Mjk3LWE2MGMtNGE4OTFhZjI3MzMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE2VDA4MjIxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1ZTAxNGM1ZDRmZjYxN2NjZWRkOGI1YTJlYThkMjNmYjlkNTI0NGMwOGFmN2VmNDgzMTYwOTg2NmZkYTJlOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CHMkupdirdWM0Q7R6Nx0QRdjDytk9xEyif7gqXNaB14"> the picture shows above is in the No.0 epoch, but the trend of loss curve seems same as above in No.20 epoch. I tried to modified the batch_size, seems not work. Also, I tried to remove/add "agent.checkpoint_path=$CHECKPOINT_PATH", but seems not work as well.
> by the way, I use 2*5090 for training. And the files are download for your relased google drive, including "depth_pretrained_v99-3jlw0p36-20210423_010520-model_final-remapped.pth", "goalflow_traj_epoch_54-step_18260.ckpt", "cluster_points_8192_.npy".
> 
> could you help me to fingure out what is the cause of this issue? Thanks in ahead for your kindly reply!

same problem with you.

**Comment by ZebinX** [comment]
@popov1212 Sorry for the delayed reply. It may happen when training the whole model directly. Have you tried first training the perception, and then training the trajectory decoder?

**Comment by popov1212** [comment]
I use your pretrained perception model and directly train the trajectory decoder

**Comment by ZebinX** [comment]
@popov1212 Is the perception module frozen or not? If the perception module is frozen, such situation could indeed occur.
